app/src/main/java/com/kamron/pogoiv/Pokefly.java

```diff
@@ -491,17 +483,73 @@ else if (possiblePoke[1] < 2) {
     /**
      * @return the likely pokemon number against the char sequence as well as the similarity
      */
-    private int[] getPossiblePokemon(CharSequence rhs) {
+    private int[] getPossiblePokemon(CharSequence poketext, CharSequence candytext) {
         int pokeNumber = 0;
         int bestMatch = 100;
-        for (int i = 0; i < pokeCalculator.pokedex.size(); i++) {
-            int similarity = pokeCalculator.get(i).getDistance(rhs);
-            if (similarity < bestMatch) {
-                pokeNumber = i;
-                bestMatch = similarity;
+
+        for (Pokemon p: pokeCalculator.pokedex) {
+            int dist = p.getDistance(poketext);
+            if (dist < bestMatch) {
+                pokeNumber = p.number;
+                bestMatch = dist;
+            }
+
+            /* We found a perfect match. We are done. */
+            if (bestMatch == 0) {
+                int[] result = {pokeNumber, bestMatch};
+                return result;
+            }
+        }
+
+        /* If we didn't find a perfect match, go on to searching by candy name.
+         * We first find the base pokemon that has the same candy name */
+        int candyNumber = 0;
+        int bestCandyMatch = 100;
+        for (Pokemon p: pokeCalculator.pokedex) {
+            /* Candy names won't match evolutions */
+            if (p.devoNumber != -1) {
+                continue;
+            }
+
+            int dist = p.getDistance(candytext);
+            if (dist < bestCandyMatch) {
+                candyNumber = p.number;
+                bestCandyMatch = dist;
             }
+            /* We found a perfect match. We are done. */
+            if (bestCandyMatch == 0) {
+                break;
+            }
+        }
+
+        /* If the pokemon name match was better than the candy name match, don't use candy name */
+        if (bestMatch < bestCandyMatch) {
```

Comment:
After @mattaereal's debugging in #129, we should think whether we want `<=` here instead of `<`. At least, the scenario where `bestMatch` and `bestCandyMatch` are both 0 (which caused the original bug) is impossible here.